tools/run_tests/xds_k8s_test_driver/framework/xds_k8s_testcase.py

```diff
@@ -316,40 +325,36 @@ def assertRpcStatusCodes(self,
                                                         after_stats)
-
         logger.info(
             '[%s] << Received accumulated stats difference.'
             ' Expecting RPCs with status %s for method %s:\n%s',
-            test_client.hostname, helpers_grpc.status_pretty(expected_status),
-            method, self._pretty_accumulated_stats(diff_stats,
-                                                   ignore_empty=True))
+            test_client.hostname, expected_status_fmt, method,
+            self._pretty_accumulated_stats(diff_stats, ignore_empty=True))
+
+        # Used in stack traces. Don't highlight for better compatibility.
+        diff_stats_fmt: str = self._pretty_accumulated_stats(diff_stats,
+                                                             ignore_empty=True,
+                                                             highlight=False)
+
+        # 1. Verify the completed RPCs of the given method has no statuses
+        #    other than the expected_status,
         stats = diff_stats.stats_per_method[method]
         for found_status_int, count in stats.result.items():
             found_status = helpers_grpc.status_from_int(found_status_int)
             if found_status != expected_status and count > stray_rpc_limit:
-                self.fail(f"Expected only status"
-                          f" {helpers_grpc.status_pretty(expected_status)},"
+                self.fail(f"Expected only status {expected_status_fmt},"
                           " but found status"
                           f" {helpers_grpc.status_pretty(found_status)}"
-                          f" for method {method}.\nStats before:"
-                          f"\n{self._pretty_accumulated_stats(before_stats)}"
-                          f"\nStats after:"
-                          f"\n{self._pretty_accumulated_stats(after_stats)}"
-                          f"\nDiff stats:"
-                          f"\n{self._pretty_accumulated_stats(diff_stats)}")
-
-        expected_status_int: int = expected_status.value[0]
-        self.assertGreater(
-            stats.result[expected_status_int],
-            0,
-            msg=("Expected non-zero RPCs with status"
-                 f" {helpers_grpc.status_pretty(expected_status)}"
-                 f" for method {method}.\nStats before:"
-                 f"\n{self._pretty_accumulated_stats(before_stats)}"
-                 f"\nStats after:"
-                 f"\n{self._pretty_accumulated_stats(after_stats)}"
-                 f"\nDiff stats:"
-                 f"\n{self._pretty_accumulated_stats(diff_stats)}"))
+                          f" for method {method}."
+                          f"\nDiff stats:\n{diff_stats_fmt}")
+
+        # 2. Verify there are completed RPCs of the given method with
+        #    the expected_status.
+        self.assertGreater(stats.result[expected_status_int],
+                           0,
+                           msg=("Expected non-zero RPCs with status"
```

Comment:
FWIW, I'd have this error be the more on-point "Expected non-zero completed RPCs for method {method}". The "with status" is actually beside the point, since there were no RPCs at all. That kept being a confusion when seeing this error.

Reply:
I agree on `completed` RPCs, however logging the status is very relevant. We're looking for different status in different tests, f.e. `DATA_LOSS` in the custom_lb test, so it's important to convey the status we were expecting.